artichoke-backend/src/extn/core/kernel/integer.rs

```diff
@@ -0,0 +1,177 @@
+use regex::Regex;
+use std::collections::HashMap;
+use std::str::FromStr;
+
+use crate::convert::{Convert, TryConvert};
+use crate::types::Ruby;
+use crate::value::Value;
+use crate::Artichoke;
+
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+pub enum Error {
```

Comment:
we should be able to make this struct `Copy` by having the `String` fields be `&'a str` instead of an owned `String`. The `&'a str` references have the same lifetime as the `Artichoke` interpreter.

Reply:
Tried to use `&'a str` but having problem passing the `digits` owned by the function as ref to `Error`, ended up with `String`. I believe I haven't fully grasp the use of lifetimes yet. 😅